### DISCUSSION TITLE
Buyer Identity - HasAnySegment In Input Function

### DISCUSSION THREAD

**Discussion by danielfv** [Shopify/function-examples#239] [Feedback: General Improvements]
Hi there,

It would be useful to have a way to check for the segments of a buyer identity in the input data of the functions.

**Comment by AndreaMorone** [comment]
Bump for this, many merchants are requesting this.

**Comment by thosakwe** [comment]
This will be great for apps that provide different discounts or behavior depending on which segment the customer is in.

**Comment by donnacrawford** [comment]
I currently need this for a number of merchants as well, where the discount function must only apply to customers within a specific segment. This is a big limitation at the moment, in the same way as if you couldn't check if a product was in a collection.

**Comment by nickwesselman** [comment]
Moving this to General Improvements since it would apply to any checkout/customer related API.

**Comment by k-ogasawara** [comment]
I'm using the Cart and Checkout Validation API in our app (https://apps.shopify.com/ruffruff-order-limits).
Currently, we need this for filtering by more complex conditions. For example, it is easy to set conditions, such as limiting the purchase of certain products, using segments of customers who bought certain products.

**Comment by gil--** [comment]
I want to add that this is useful for B2B use-cases since Companies don't support tags it would be nice to bulk query Customers based on their Company via Segments.

**Comment by thosakwe** [comment]
Here's why this feature request is important:

* Giving merchants the ability to create more segmented offers would help them maximize revenue.
* Being able to read customer segment memberships in a Function would reduce the impact of the input query complexity limit. For example, instead of requesting multiple tags and metafields, Functions could query a single segment and only consume 3 points of the limit.

Please consider adding `Customer.inSegment` and `Customer.inAnySegment` fields to your roadmap. Thanks!

**Comment by blanklob** [comment]
I totaly agree here, this has huge impact and would reduce so much complexity with input.

**Comment by dfmcphee** [comment]
This would be a big help for us too. We also want to be able to gate discounts and function logic on whether a customer belongs to specific segments.
